ddl/partition.go

```diff
@@ -1335,6 +1335,12 @@ func checkExchangePartitionRecordValidation(w *worker, pt *model.TableInfo, inde
 		} else if len(pi.Columns) == 1 {
 			sql = buildCheckSQLForRangeColumnsPartition(pi, index, schemaName, tableName)
 		}
+	case model.PartitionTypeList:
+		if len(pi.Columns) == 0 {
+			sql = buildCheckSQLForListPartition(pi, index, schemaName, tableName)
+		} else {
```

Comment:
Lack a check `if len(pi.Columns) == 1`, you are assuming `colName := pi.Columns[0].L`, that `[0]` index in `buildCheckSQLForListColumnsPartition`.

Reply:
OK, I have add this check.